File: VB/WindowsApplication54/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Data.OleDb
Imports System.Windows.Forms
Imports DevExpress.XtraReports
Imports DevExpress.XtraReports.Extensions
Imports DevExpress.XtraReports.UI
' ...

Namespace WindowsApplication54
    Partial Public Class Form1
        Inherits Form
        Shared Sub New()
            ' The following code is required to support serialization of multiple custom objects.
            TypeDescriptor.AddAttributes(GetType(DataSet), New ReportAssociatedComponentAttribute())
            TypeDescriptor.AddAttributes(GetType(OleDbDataAdapter), New ReportAssociatedComponentAttribute())

            ' The following code is required to serialize custom objects.
            ReportExtension.RegisterExtensionGlobal(New ReportExtension())
            ReportDesignExtension.RegisterExtension(New DesignExtension(), ExtensionName)
        End Sub
        Private Const ExtensionName As String = "Custom"

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub createReportWhithDataSourceButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles createReportWhithDataSourceButton.Click
            Using report As New XtraReport()
                Using tool As New ReportDesignTool(report)
                    AddHandler tool.DesignForm.DesignMdiController.DesignPanelLoaded, AddressOf OnDesignPanelLoaded
                    tool.ShowDesignerDialog()
                End Using
            End Using
        End Sub

        Private Sub OnDesignPanelLoaded(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs)
            ReportDesignExtension.AssociateReportWithExtension(CType(e.DesignerHost.RootComponent, XtraReport), ExtensionName)
        End Sub

        Private Sub loadReportfromFileButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles LoadReportfromFileButton.Click
            Dim openfd As New OpenFileDialog()
            If openfd.ShowDialog() <> System.Windows.Forms.DialogResult.OK Then
                Return
            End If
            Dim report As New XtraReport()
            report.LoadLayoutFromXml(openfd.FileName)
            report.ShowDesignerDialog()
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication54/Form1.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Data.OleDb;
using System.Windows.Forms;
using DevExpress.XtraReports;
using DevExpress.XtraReports.Extensions;
using DevExpress.XtraReports.UI;
// ...

namespace WindowsApplication54 {
    public partial class Form1 : Form {
        static Form1() {
            // The following code is required to support serialization of multiple custom objects.
            TypeDescriptor.AddAttributes(typeof(DataSet), new ReportAssociatedComponentAttribute());
            TypeDescriptor.AddAttributes(typeof(OleDbDataAdapter), new ReportAssociatedComponentAttribute());

            // The following code is required to serialize custom objects.
            ReportExtension.RegisterExtensionGlobal(new ReportExtension());
            ReportDesignExtension.RegisterExtension(new DesignExtension(), ExtensionName);
        }
        private const string ExtensionName = "Custom";

        public Form1() {
            InitializeComponent();
        }
        private void createReportWhithDataSourceButton_Click(object sender, EventArgs e) {
            using(XtraReport report = new XtraReport()) {
                using(ReportDesignTool tool = new ReportDesignTool(report)) {
                    tool.DesignForm.DesignMdiController.DesignPanelLoaded += OnDesignPanelLoaded;
                    tool.ShowDesignerDialog();
                }
            }
        }
        void OnDesignPanelLoaded(object sender, DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs e) {
            ReportDesignExtension.AssociateReportWithExtension((XtraReport)e.DesignerHost.RootComponent, ExtensionName);
        }
        private void loadReportfromFileButton_Click(object sender, EventArgs e) {
            OpenFileDialog openfd = new OpenFileDialog();
            if (openfd.ShowDialog() != DialogResult.OK)
                return;
            XtraReport report = new XtraReport();
            report.LoadLayoutFromXml(openfd.FileName);
            report.ShowDesignerDialog();
        }
    }
}